### PR TITLE
added routes/view to user/product show

### PR DESCRIPTION
**Pull request by iamsammysam**
Added routes/made changes to product controller to show merchant view of a product when logged in. Working on product show view for all other customers.

### CODE REVIEW

File: app/views/products/index.html.erb [hunk 1]
```diff
@@ -11,7 +11,7 @@
         <li>Stock: <%= product.stock %> units</li>
         <li>Available: <%= product.available %></li>
         <li>User_id: <%= product.user_id %></li>
-        <img src= <%= product.photo_url %>
+        <img src= <%= product.photo_url %> alt=<% product.name %> height="200" width="200">
```
Comment (Galaxylaughing) [inline app/views/products/index.html.erb]: I think for the alt text you need to use `<%=`, but I'm not sure.

File: app/controllers/products_controller.rb [hunk 2]
```diff
@@ -16,6 +15,10 @@ def show
       redirect_to products_path
       return
     end   
+    
+    if session[:uid]
+      render :user_show
+    end
```
Comment (Galaxylaughing) [inline app/controllers/products_controller.rb]: I mean, if it works, it works, but I'm confused: where are we setting `session[uid]`? I know when someone logs in, it sets `session[:user_id]`, but wouldn't we also need to check that the product page the logged-in user is trying to view actually belongs to them and not another user?

Is that checking being done in some other method that then sets `session[:uid]`?

File: app/views/products/index.html.erb (hunk 1 above)
Comment (iamsammysam) [reply app/views/products/index.html.erb]: Done!

File: app/views/products/user_show.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<body>
```
[the remark this answers is not in the document]
Comment (iamsammysam) [reply app/views/products/user_show.html.erb]: Ok!

**Pull request merged**